packages/components/src/text-control/style.scss

```diff
@@ -20,5 +20,7 @@
 
 	&.is-next-40px-default-size {
 		height: $grid-unit-50;
+		padding-inline-start: $grid-unit-20 - 1px;
+		padding-inline-end: $grid-unit-20 - 1px;
```

Comment:
Subtracting 1px to account for the fact that the newer components do not include a border on the underlying element. Also tweaked to use logical properties (though it doesn't really matter here), since we'll likely start linting for that in this package.

```diff
diff --git a/packages/components/src/input-control/stories/index.story.tsx b/packages/components/src/input-control/stories/index.story.tsx
index 6067c467f9..0b3353e0c7 100644
--- a/packages/components/src/input-control/stories/index.story.tsx
+++ b/packages/components/src/input-control/stories/index.story.tsx
@@ -14,6 +14,9 @@ import InputControl from '..';
 import { InputControlPrefixWrapper } from '../input-prefix-wrapper';
 import { InputControlSuffixWrapper } from '../input-suffix-wrapper';
 import Button from '../../button';
+import TextControl from '../../text-control';
+import SelectControl from '../../select-control';
+import { VStack } from '../../v-stack';
 
 const meta: Meta< typeof InputControl > = {
 	title: 'Components (Experimental)/InputControl',
@@ -112,3 +115,23 @@ export const ShowPassword: StoryFn< typeof InputControl > = ( args ) => {
 		/>
 	);
 };
+
+export const SizeTest = {
+	render: () => {
+		return (
+			<VStack spacing={ 2 }>
+				<InputControl __next40pxDefaultSize value="Hello" />
+				<TextControl
+					__nextHasNoMarginBottom
+					__next40pxDefaultSize
+					value="Hello"
+					onChange={ () => null }
+				/>
+				<SelectControl
+					__next40pxDefaultSize
+					options={ [ { label: 'Hello', value: 'hello' } ] }
+				/>
+			</VStack>
+		);
+	},
+};

```

Reply:
What's the difference between `InputControl` and `TextControl`?

Reply:
Should we add a comment to explain the need for subtracting 1px?

Reply:
> ```diff
> InputControl
> ```

`InputControl` is newer, still formally experimental, and has a few more features that `TextControl` doesn't  - like prefix, suffix, and additional ways to display the label. Check the storybook stories to get a better idea.

Reply:
> InputControl is newer, still formally experimental, and has a few more features that TextControl doesn't - like prefix, suffix, and additional ways to display the label. Check the storybook stories to get a better idea.

Why didn't we just update TextControl with the new features (and new implementation)

Reply:
Comment added in 829e08277452cd2230dc4009ce0d16b8cfaf8d43

Reply:
I wasn't part of the team back then, but IIRC `InputControl` came along with G2, and was aiming to be `TextControl`'s successor. My assumption is that we preferred a new component in order to supply all the new features (including making it themeable) without breaking any backwards compatibility. It was also intended for those 2 components to converge, but it never happened. It might be something to pursue at some point, and it's possible that others might have thought about it already (@WordPress/gutenberg-components)

Reply:
The converging could've happened as part of #59418, but due to the Emotion performance issues we found in https://github.com/WordPress/gutenberg/pull/58948#issuecomment-1942040476, I'd want to wait until we remove Emotion from InputControl. It is definitely in the plans though, since it doesn't make sense to have two input components in the package.